### ISSUE TITLE
Process Guide: add links to documentation and examples for using Travis CI

### ISSUE THREAD

**Issue by mdejongh** [kbase/project_guides#26]
The Process Guide is very clear and helpful, and the proposal for extensive unit, integration and systems testing is excellent. It would be helpful to have some links to documentation for how to install and configure Travis CI for a GitHub repo, as well as an example of a KBase repo that has successfully set up Travis CI.

**Comment by cshenry** [comment]
Yeah, matt really hit�s the nail on the head with his comment. I understand the broad guidelines, but I could use a clearer understand of how to put these things in action. I see some actions to implement some of these policies else where, but I�m missing the forest for the trees. What is the broader vision here?

**Comment by fperez** [comment]
I think there are two questions above, if I understand correctly:
1. what's the broader vision
2. specific links to examples and guidance.

For `#2`, point well taken: I'll leave this issue open and have assigned it to myself so I don't forget.

For `#1`, the vision is that by having coherent, pervasive testing at multiple layers (from simple Travis-driven tests for every PR based on per-repo unit tests to more complex nightly system-wide integration tests on our own infrastructure, likely with Jenkins), we will over time progress towards a much more comprehensible and robust codebase.  We won't get there overnight, but we will for sure _never_ get there if we don't start taking pervasive testing seriously.

**Comment by briehl** [comment]
I've worked through how to set up Travis-CI and Coveralls on a personal repo the other day, and plan to integrate these into the Narrative repo soon (probably one of the more complicated, due to multiple code languages that need to be covered).

I'll be happy to write up what I did and put that document in the project guide, with links to external tutorials. It's a little different for different languages, but not really crazy.

In the meantime, the [Narrative Method Store](https://github.com/kbase/narrative_method_store) and [Data API](https://github.com/kbase/data_api) repos are set up with Travis-CI, so you can see what the final product looks like, but that doesn't really inform how to get there.
